set.mm

```diff
@@ -507514,6 +507574,181 @@ its graph has a given second element (that is, function value).
+  sgnnbi $p |- ( A e. RR* -> ( ( sgn ` A ) = -u 1 <-> A < 0 ) ) $=
+    ( cxr wcel csgn cfv c1 cneg wceq cc0 clt wi id eqeq1 imbi1d neg1ne0 pm2.21i
+    wbr wa a1i neg1rr nesymi neg1lt0 0lt1 0re 1re lttri mp2an gtneii neii simp2
+    3expia sgn3da imp sgnn impbida ) ABCZADEZFGZHZAIJQZUPUSUTUPUSUTKIURHZUTKZFU
+    RHZUTKZURURHZUTKAUPLUQIHUSVAUTUQIURMNUQFHUSVCUTUQFURMNUSUSVEUTUQURURMNVBUPA
+    IHRVAUTURIOUAPSVDUPIAJQRVCUTFURURFTURIJQIFJQURFJQUBUCURIFTUDUEUFUGUHUIPSUPU
+    TVEUTUPUTVEUJUKULUMAUNUO $.
+
+  $( Positive signum.  (Contributed by Thierry Arnoux, 2-Oct-2018.) $)
+  sgnpbi $p |- ( A e. RR* -> ( ( sgn ` A ) = 1 <-> 0 < A ) ) $=
+    ( cxr wcel csgn cfv c1 wceq cc0 clt wi cneg id eqeq1 imbi1d wa 0ne1 pm2.21i
+    wbr a1i neg1rr neii simp2 3expia neg1lt0 0lt1 0re lttri mp2an gtneii nesymi
+    1re sgn3da imp sgnp impbida ) ABCZADEZFGZHAIRZUPURUSUPURUSJHFGZUSJZFFGZUSJF
+    KZFGZUSJZAUPLUQHGURUTUSUQHFMNURURVBUSUQFFMNUQVCGURVDUSUQVCFMNVAUPAHGOUTUSHF
+    PUAQSUPUSVBUSUPUSVBUBUCVEUPAHIROVDUSFVCVCFTVCHIRHFIRVCFIRUDUEVCHFTUFUKUGUHU
+    IUJQSULUMAUNUO $.
+
+  $( Zero signum.  (Contributed by Thierry Arnoux, 10-Oct-2018.) $)
+  sgn0bi $p |- ( A e. RR* -> ( ( sgn ` A ) = 0 <-> A = 0 ) ) $=
+    ( cxr wcel csgn cfv cc0 wceq wb c1 cneg eqeq1 bibi1d wa clt wbr ax-1ne0 a1i
+    simpr neneqd 2falsed eqcomd eqeq1d wne gt0ne0d 1cnd negne0d lt0ne0d sgn3da
+    id ) ABCZADEZFGZAFGZHFFGZUMHIFGZUMHIJZFGZUMHAUJUIULULUNUMUKFFKLUKIGULUOUMUK
+    IFKLUKUPGULUQUMUKUPFKLUJUMMZFAFURAFUJUMRUAUBUJFANOZMZUOUMUTIFIFUCZUTPQSUTAF
+    UTAUJUSRUDSTUJAFNOZMZUQUMVCUPFVCIVCUEVAVCPQUFSVCAFVCAUJVBRUGSTUH $.
+
+  $( Signum is idempotent.  (Contributed by Thierry Arnoux, 2-Oct-2018.) $)
+  sgnsgn $p |- ( A e. RR* -> ( sgn ` ( sgn ` A ) ) = ( sgn ` A ) ) $=
+    ( cxr wcel csgn cfv wceq cc0 c1 cneg id fveq2 eqeq12d sgn0 a1i clt wbr sgn1
+    wa neg1rr rexri neg1lt0 sgnn mp2an sgn3da ) ABCZADEZDEZUFFGDEZGFZHDEZHFZHIZ
+    DEZULFZAUEJUFGFZUGUHUFGUFGDKUOJLUFHFZUGUJUFHUFHDKUPJLUFULFZUGUMUFULUFULDKUQ
+    JLUIUEAGFRMNUKUEGAOPRQNUNUEAGOPRULBCULGOPUNULSTUAULUBUCNUD $.
+
+  $( If two real numbers are of different signs, so are their signs.
+     (Contributed by Thierry Arnoux, 12-Oct-2018.) $)
+  sgnmulsgn $p |- ( ( A e. RR /\ B e. RR )
+    -> ( ( A x. B ) < 0 <-> ( ( sgn ` A ) x. ( sgn ` B ) ) < 0 ) ) $=
+    ( cr wcel wa cmul co csgn cfv c1 cneg wceq cc0 clt wbr neg1lt0 breq1 mpbiri
+    simpr simplr adantl lt0ne0d pm2.21ddne eqbrtrrd cn0 wn nn0nlt0 pm2.21dd cxr
+    1nn0 mp1i ctp w3o remulcl rexrd adantr sgncl eltpi mpjao3dan impbida sgnnbi
+    3syl wb syl sgnmul breq1d 3bitr3d ) ACDBCDEZABFGZHIZJKZLZVJMNOZVIMNOZAHIBHI
+    FGZMNOVHVLVMVLVMVHVLVMVKMNOPVJVKMNQRUAVHVMEZVLVLVJMLZVJJLZVPVLSVPVQEZVLVJMV
+    PVQSVSVJVHVMVQTUBUCVPVREZJMNOZVLVTVJJMNVPVRSVHVMVRTUDJUEDWAUFVTUJJUGUKUHVPV
+    IUIDZVJVKMJULDVLVQVRUMVHWBVMVHVIABUNUOZUPVIUQVJVKMJURVBUSUTVHWBVLVNVCWCVIVA
+    VDVHVJVOMNABVEVFVG $.
+
+  $( If two real numbers are of different signs, so are their signs.
```

Comment:
"of same signs"?

Reply:
Fixed in https://github.com/metamath/set.mm/pull/5093/commits/3a25d71a71e4a6c5620e86f7d62c8a5fa088d0ef